stories/core/Modal.stories.tsx

```diff
@@ -110,15 +113,213 @@ export const Basic: Story<ModalProps> = ({
   );
 };
 
-export const NonDismissibleModal: Story<ModalProps> = Basic.bind({});
+export const NonDismissibleModal: Story<ModalProps> = ({
+  isOpen,
+  title = 'This is the title',
+  ...rest
+}) => {
+  const [isModalOpen, setIsModalOpen] = useState(isOpen);
+
+  useEffect(() => {
+    setIsModalOpen(isOpen);
+  }, [isOpen]);
+
+  const closeModal = () => {
+    setIsModalOpen(false);
+  };
+
+  const onClose = (event: React.SyntheticEvent<Element, Event>) => {
+    action('onClose', { depth: 1 })(event);
+    closeModal();
+  };
+
+  const primaryButtonHandle = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    action('Primary button')(event);
+    closeModal();
+  };
+
+  const secondaryButtonHandle = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    action('Secondary button')(event);
+    closeModal();
+  };
+
+  return (
+    <>
+      <Button styleType='high-visibility' onClick={() => setIsModalOpen(true)}>
+        Open Modal
+      </Button>
+      <Modal
+        isOpen={isModalOpen}
+        title={title}
+        onClose={onClose}
+        onKeyDown={action('onKeyDown', { depth: 1 })}
```

Comment:
Now its fixed, great finding!!
One minor thing...
Could you also repeat the props specific per story in here?
If you look at storybook, it is not clear, what kinda props were passed to have non dismissible modal, for example.
The ones we set in args, are not really visible in the code.
So in this story just add `isDismissible={false}` just before `{...rest}`

And in other same (in full page you would add `styleType` explicitly, for example).

Reply:
Ah, of course. I wanted it to show up in the controls. Turns out you can do both.